File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        l=0
        r=0
        maxFreq=0
        windowSum=0

        for r in range (len(nums)):
            windowSum+=nums[r]
            while(nums[r]*(r-l+1) > windowSum+k):
                windowSum-=nums[l]
                l+=1
            
            maxFreq = max(maxFreq, r-l+1)
        
        return maxFreq
        

File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        long long int l=0, r=0, maxFreq=0, sum=0, n=nums.size();
        for(r;r<n;r++){
            sum+=nums[r];
            if(l+1<=r && nums[r]*(r-l+1) > sum+k){
                sum=sum-nums[l];
                l++;
            }
            maxFreq = max(maxFreq, r-l+1);
        }
        return maxFreq;
    }
};